### ISSUE TITLE
Output Vanilla SegNet

### ISSUE THREAD

**Issue by BarbeBleue** [j96w/DenseFusion#38]
I'm trying to use the output of vanilla SegNet network to label YCB-Video images but I don't find an efficient way to transform the 22\*640\*480 output into a single label image 640*480.
For the moment I'm using something like that:

```
seg_data = seg(rgb) #  output SegNet
seg_data = seg_data.detach().cpu().numpy()[0]
seg_image = np.zeros((480, 640))
obj_list = []
for i in range(480):
    for j in range(640):
        prob_max = 0
        label = 0
        for r in range(22):
            if seg_data[r][i][j] > prob_max:
                label = r
                prob_max = seg_data[r][i][j]
        seg_image[i][j] = label
        if label not in obj_list:
            obj_list.append(label)
```

How do you use the output for fast segmentation of an rgb image ?

**Comment by BarbeBleue** [comment]
I found a faster way but I still don't know if it's the best one:
```
seg_data = seg_data[0]
seg_data2 = torch.transpose(seg_data2, 0, 2)
seg_data2 = torch.transpose(seg_data2, 0, 1)
seg_image = torch.argmax(seg_data2, dim=-1)
obj_list = torch.unique(seg_image).detach().cpu().numpy()
seg_image = seg_image.detach().cpu().numpy()
```

**Comment by j96w** [comment]
Similar to mine, actually.
